src/main/java/edu/harvard/iq/dataverse/api/Datasets.java

```diff
@@ -3917,4 +3923,32 @@ public Response getDatasetVersionCitation(@Context ContainerRequestContext crc,
         return response(req -> ok(
                 getDatasetVersionOrDie(req, versionId, findDatasetOrDie(datasetId), uriInfo, headers).getCitation(true, false)), getRequestUser(crc));
     }
+
+    @PUT
+    @AuthRequired
+    @Path("{id}/versions/{versionId}/deaccession")
+    public Response deaccessionDataset(@Context ContainerRequestContext crc, @PathParam("id") String datasetId, @PathParam("versionId") String versionId, String jsonBody, @Context UriInfo uriInfo, @Context HttpHeaders headers) {
+        if (DS_VERSION_DRAFT.equals(versionId) || DS_VERSION_LATEST.equals(versionId)) {
```

Comment:
If the "latest" is published would this be ok?

Reply:
Since ":latest" considers daft version (If you have permissions), I opted for restricting ":latest" and considered only ":latest-published". 

Conceptually I think it makes sense, and makes the underlaying logic simpler since otherwise we would have to add additional logic in the dataset version recovery commands (for latest) when we search for a dataset for deaccessioning.

If you think it is relevant enough, I can update it and add such logic.

Reply:
Just to make it easier on the user you could allow them to use "latest", do the getDraftVersionOrDie first and check to see if that version is draft. Also, do we want to check to see if the version is already deaccessioned? Maybe it's ok and we want to allow them to update the reason/url.

Reply:
With respect to re-deaccessioning, it's not allowed in the UI.

Reply:
I'm fine with only supporting ":latest-published". Only published datasets can be deaccessioned, after all. If there's a strong need, we could always add ":latest" later.

Reply:
The drawback I see with :latest is that the dataset it refers to depends on the user's permissions. If the user can access the draft version, :latest will correspond to draft (Current behavior), otherwise it will search for the latest published version if there is one.

The problem is that you can't deaccession a draft version. Therefore it can be a bit confusing for the user. An alternative I thought of is to never point to the draft version for this particular endpoint when using :latest, but this can also create confusion for the user as the other endpoints would continue to support draft for :latest.

Reply:
My suggestion was to allow the use of latest and if that happens to be draft to give an error response. We can leave it as only allow latest-published.